### PR TITLE
[Unity][BYOC] CoreML Scaffolding

### PR DESCRIPTION
**Pull request by sunggg**
This PR lands P4 of [CoreML tracker](https://github.com/apache/tvm/issues/15245). 

Although it produces the right results, its CPU performance is currently quite worse than expected on my Macbook with Intel chips. 
For more performance analysis on more recent Apple silicon, I'd like to land the functionality first and then iterate over to identify the performance issue as the next step (P5).

### CODE REVIEW

File: tests/python/relax/test_codegen_coreml.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    ex1 = relax.build(mod1, target=target)
+    vm1 = relax.VirtualMachine(ex1, dev, profile=True)
+    out1 = vm1["main"](*inputs)
+
+    mod2 = relax.transform.LegalizeOps()(mod)
+    ex2 = relax.build(mod2, target=target)
+    vm2 = relax.VirtualMachine(ex2, dev, profile=True)
+    out2 = vm2["main"](*inputs)
+
+    tvm.testing.assert_allclose(out1.numpy(), out2.numpy(), rtol=1e-3, atol=1e-3)
```
Comment (sunggg) [inline tests/python/relax/test_codegen_coreml.py:60]: Please note that current error margin is bit higher than our usual setting for CPU computations (`rtol=1e-7, atol=1e-7`), but lower than the one for GPU libraries (cutlass: `rtol=1e-2, atol=1e-2`)

**Comment by tqchen** [comment]
@vinx13 @yelite do you mind take a look at this pr

File: src/relax/transform/run_codegen.cc [hunk 2]
```diff
@@ -86,14 +86,17 @@ class CodeGenRunner : ExprMutator {
       };
 
       if (auto it = extern_funcs_.find(gvar_node); it != extern_funcs_.end()) {
-        return create_call_dps_packed(it->second.first, it->second.second);
+        auto call = create_call_dps_packed(it->second.first, it->second.second);
+        LOG(INFO) << call;
```
[the remark this answers is not in the document]
Comment (sunggg) [reply src/relax/transform/run_codegen.cc]: Thank you for the catch! This is fixed.

**Review by yelite — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**